pkg/spanconfig/spanconfigstore/store.go

```diff
@@ -206,13 +209,26 @@ func (s *Store) ForEachOverlappingSpanConfig(
 ) error {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
-	return s.mu.spanConfigStore.forEachOverlapping(span, func(sp roachpb.Span, conf roachpb.SpanConfig) error {
+	foundNoOverlapping := true
+	err := s.mu.spanConfigStore.forEachOverlapping(span, func(sp roachpb.Span, conf roachpb.SpanConfig) error {
+		foundNoOverlapping = false
 		config, err := s.getSpanConfigForKeyRLocked(ctx, roachpb.RKey(sp.Key))
 		if err != nil {
 			return err
 		}
 		return f(sp, config)
 	})
+	if err != nil {
+		return err
+	}
+	if foundNoOverlapping {
```

Comment:
Definitely worth a comment, both for how we get here and what this block is doing: namely that getSpanConfigForKeyRLocked returns config that is combined with the system span config so that this ensures the system span config is respected by callers. We might change the function level comment to indicate that the provided func will always be called at least once unless an error is returned.

I also wonder what KV would think of an `info.VInfo(ctx, 3, ...` here.